File: leetcode/657/Robot.java
class Robot
{
    public static void main(String[] args) {
        Solution sl = new Solution();
        System.out.println(sl.judgeCircle("UD"));
    }
}
class Solution {
    public boolean judgeCircle(String moves) {
        int x = 0;
        int y = 0;
        
        char[] array = moves.toCharArray();
        
        for(int i = 0; i < array.length; i++)
        {
            if(array[i] == 'R')
                x++;    
            else if(array[i] == 'L')
                x--;
            else if(array[i] == 'U')
                y++;    
            else if(array[i] == 'D')
                y--;
        }
        return (x == 0 && y == 0);
    }
}

File: leetcode/657/Robot.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>
bool judgeCircle(char * moves){
    
    int length = strlen(moves);
    
    int x = 0;
    int y = 0;
    int i = 0;
    
    for(i = 0; i < length; i++)
    {
        if(moves[i] == 'R')
            x++;    
        else if(moves[i] == 'L')
            x--;
        else if(moves[i] == 'U')
            y++;    
        else if(moves[i] == 'D')
            y--;
    }
    return (x == 0 && y == 0);
}

int main() {
 printf(judgeCircle("UD")?"true\n":"false\n");   
 return 0;
}